packages/core/src/directory.ts

```diff
@@ -0,0 +1,263 @@
+import { Org, OrgID, OrgMember, OrgMemberStatus, Group } from "./org";
+import { Server } from "./server";
+import { getIdFromEmail } from "./util";
+import { Auth } from "./auth";
+
+export interface DirectoryUser {
+    externalId?: string;
+    email?: string;
+    name?: string;
+    active?: boolean;
+}
+
+export interface DirectoryGroup {
+    externalId?: string;
+    name: string;
+    members?: DirectoryUser[];
+}
+
+export interface DirectoryGroupChanges {
+    newName?: string;
+    memberIdsToAdd?: string[];
+    memberIdsToRemove?: string[];
+}
+
+export interface DirectorySubscriber {
+    userCreated(user: DirectoryUser, orgId: OrgID): Promise<OrgMember | void>;
+    userUpdated(user: DirectoryUser, orgId: OrgID, userId: string): Promise<OrgMember | void>;
```

Comment:
I don't think we need the third `userId` argument, since that is what `DirectoryUser.externalId` is for.

Reply:
`externalId` is supposed to be given to us (provider), not generated by us, as per my understanding of the RFC: https://datatracker.ietf.org/doc/html/rfc7643#section-3.1 so we need the `DirectoryUser.id` which is what I was using as `userId` here.